src/runtime_src/core/common/usage_metrics.cpp

```diff
@@ -0,0 +1,515 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2023 Advanced Micro Devices, Inc. All rights reserved.
+#define XRT_API_SOURCE
+#define XCL_DRIVER_DLL_EXPORT
+#define XRT_CORE_COMMON_SOURCE
+#include "config_reader.h"
+#include "usage_metrics.h"
+
+#include "core/common/api/hw_context_int.h"
+#include "core/common/api/kernel_int.h"
+#include "core/common/device.h"
+#include "core/common/query.h"
+#include "core/common/query_requests.h"
+#include "core/common/shim/buffer_handle.h"
+#include "core/common/shim/hwctx_handle.h"
+#include "core/include/xrt/xrt_uuid.h"
+
+#include <algorithm>
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/property_tree/ptree.hpp>
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <mutex>
+#include <sstream>
+#include <thread>
+
+#ifdef _WIN32
+# pragma warning ( disable : 4996 )
+#endif
+
+namespace bpt = boost::property_tree;
+namespace {
+// global variables
+std::mutex m;
+uint32_t thread_count;
+} // namespace
+
+namespace xrt_core::usage_metrics {
+
+template <typename met_type, typename find_type>
+static met_type*
+get_metrics(std::vector<met_type>& metrics_vec, const find_type& finder)
+{
+  auto it = std::find_if(metrics_vec.begin(), metrics_vec.end(), 
+                        [finder](met_type& met) 
+                        {
+                          return met.handle == finder;
+                        }
+                        );
+  return it == metrics_vec.end()
+    ? nullptr
+    : &(*it);
+}
+
+struct bo_metrics
+{
+  uint32_t total_count = 0;
+  size_t   total_size_in_bytes = 0;
+  size_t   peak_size_in_bytes = 0;
+  size_t   bytes_synced_to_device = 0;
+  size_t   bytes_synced_from_device = 0;
+};
+
+using tp = std::chrono::time_point<std::chrono::high_resolution_clock>;
+struct kernel_metrics
+{
```

Comment:
Scope to where it is used
```suggestion
struct kernel_metrics
{
  using tp = std::chrono::time_point<std::chrono::high_resolution_clock>;
```